### DISCUSSION TITLE
Warm starting Lagrange Multipliers

### DISCUSSION THREAD

**Discussion by sneukirch** [coin-or/Ipopt#550] [Q&A] [answered]
dear all,
I am using Ipopt for a structural mechanics problem. It is working nicely and efficiently.
Basically, I compute the stable equilibrium configuration of an elastic beam (minimizing the elastic energy) for a given load value (lambda=1.0). Then I increase the load value (lambda=1.1) and compute the next equilibrium configuration. When computing this new equilibrium config (with lambda=1.1), I warm-start Ipopt with the former configuration (with lambda=1.0). All this is good and we even manage to warm-start the Lagrange multipliers (associated with equality constraints). We also warm-start the Kuhn multipliers (associated with inequality constraints), but here comes my problem: there seems to be something we do not manage to warm-start because Ipopt uses too many iterations. For example, if I compute the configuration for lambda=1.0, then I start Ipopt again with the same lambda=1.0 value, Ipopt is going to take 10 iterations to converge again on the same solution. My 2-cents guess is that the mu for the "log-barrier" associated with inequality constraints is not warm-started, but I would be happy to have a more detailed explanation on this. 
Happy 2022!
sebastien

**Comment by svigerske** [comment, marked as answer]
#529 could be relevant.
Also https://list.coin-or.org/pipermail/ipopt/2016-May/004221.html

**Reply by sneukirch** [reply]
Thank you Stefan
I will look at these...
